ircd/s_user.c

```diff
@@ -1469,6 +1474,29 @@ oper_up(struct Client *source_p, struct oper_conf *oper_p)
 	send_umode_out(source_p, source_p, old);
 	sendto_one_numeric(source_p, RPL_SNOMASK, form_str(RPL_SNOMASK),
 		   construct_snobuf(source_p->snomask));
+
+	if (HasPrivilege(source_p, "auspex:cmodes"))
```

Comment:
You don't `oper_up` when your privset changes, so I think this is the wrong place to do anything that checks privs. Currently the best place is probably `user_mode`, hacky though it is, since we arrange to call it every time privs change.

Reply:
doesn't it also get called for other stuff? how are we meant to know when `/mode mynick` isn't supposed to send out these `MODE`s?

Reply:
I don't think we can :(

Maybe it's time to bite the bullet and add a standard mechanism for all privilege changes

Reply:
what does this involve